plugins/modules/fusion_ra.py

```diff
@@ -117,12 +135,21 @@ def user_to_principal(fusion, user_id):
     principal = None
     id_api_instance = purefusion.IdentityManagerApi(fusion)
     users = id_api_instance.list_users()
-    for user in range(0, len(users)):
-        if users[user].name == user_id:
-            principal = users[user].id
+    for user in users:
+        if user.name == user_id:
+            principal = user.id
     return principal
 
 
+def apiclient_to_principal(module, fusion, api_client_key):
+    """Given an API client key, such as "pure1:apikey:123xXxyYyzYzASDF" (also known as issuer_id),
+    return the associated principal
+    """
+    id_api_instance = purefusion.IdentityManagerApi(fusion)
+    api_clients = id_api_instance.list_users(name=api_client_key)
+    return api_clients[0].id
```

Comment:
This will raise an exception if the list is empty. Add check that the list is not empty.

Reply:
fixed in [f12771f](https://github.com/Pure-Storage-Ansible/Fusion-Collection/pull/138/commits/f12771fc26277a43357cbbbe0790ceabcdbad6d7)